src/OrleansRuntime/GrainTypeManager/GrainTypeData.cs

```diff
@@ -151,5 +154,11 @@ internal static PlacementStrategy GetPlacementStrategy(Type grainClass, Placemen
 
             return predicate;
         }
+
+        private ParameterInfo[] GetConstructorParameters()
+        {
+            var constructor = this.Type.GetConstructors(BindingFlags.Public | BindingFlags.Instance).FirstOrDefault();
+            return constructor?.GetParameters() ?? EmptyParameters;
+        }
```

Comment:
"Made it to something more core to Orleans"

As an extensibility pattern, it needs to be core to Orleans.  I'd argue we want the pattern to be uniform and formalized.  We don't want different facets using different patterns that only work with some containers.  Custom application specific facets can of course be generated, but those will be application specific and will not live in our tree.  Note that this change does not prevent those, only formalizes what we support.

"IMO still needs more thought with regards to 3rd party containers"
Agreed, and this is subject to change.  However, whatever we come up with should be the expected and supported pattern.  If application code deviates, that's an application problem.

"I'd couple it to DefaultGrainActivator"
I think we should think on this.  I'd originally suggested that capability be part of the activator, and there was push back because it broke the asp.net pattern that was being used.  The entire reason we're using this pattern to bind to parameters, rather than the original pattern of passing the constructed facets to the constructor was that we wanted to preserve the asp.net patterns, and now we're discussing deviating from this for what is being described as a caching optimization?